cranelift/src/run.rs

```diff
@@ -85,12 +85,11 @@ fn run_file_contents(file_contents: String) -> Result<()> {
         ..ParseOptions::default()
     };
     let test_file = parse_test(&file_contents, options)?;
-    let isa = create_target_isa(&test_file.isa_spec)?;
-    let mut compiler = SingleFunctionCompiler::new(isa);
     for (func, Details { comments, .. }) in test_file.functions {
         for comment in comments {
             if let Some(command) = parse_run_command(comment.text, &func.signature)? {
-                let compiled_fn = compiler.compile(func.clone())?;
+                let isa = create_target_isa(&test_file.isa_spec)?;
+                let compiled_fn = SingleFunctionCompiler::new(isa).compile(func.clone())?;
```

Comment:
I'd prefer not to move this code into the loop, if my earlier suggestion on shared references to the `TargetIsa` works out.